### PR TITLE
feat(api-gateway,event-bridge): apigateway to event bus put events integration

### PR DESCRIPTION
**Pull request by thantos**
Depends on #347

* Support sending API gatgeway requests to Event Bridge
    * Caveat: only supports object literals
* Bug: API Gateway integration strings in SFN were incorrectly stringified (from #361).

### CODE REVIEW

File: src/event-bridge/event-bus.ts [hunk 1]
```diff
@@ -345,6 +356,100 @@ abstract class EventBusBase<in Evnt extends Event, OutEvnt extends Evnt = Evnt>
...
+              throw new SynthError(
+                ErrorCodes.Expected_an_object_literal,
+                "API Gateway Integration with EventBus.putEvents expects object literals with no computed properties"
+              );
+            }
+
+            const objectProps = arg.properties.map((prop) => {
+              if (
+                !isPropAssignExpr(prop) ||
+                !(isIdentifier(prop.name) || isStringLiteralExpr(prop.name))
```
Comment (sam-goodwin) [inline src/event-bridge/event-bus.ts:386]: nit: are the parens necessary?

Comment (sam-goodwin) [reply src/event-bridge/event-bus.ts:386]: Would it be easier to check for what's allowed instead of the `!`? I am confused.

Comment (sam-goodwin) [reply src/event-bridge/event-bus.ts:386]: what about numeric properties?

File: test/api.test.ts [hunk 2]
```diff
@@ -213,6 +214,63 @@ test("AWS integration with Standard Step Function using input data", () => {
...
+            body: {
+              event: { source: string; "detail-type": string; detail: any };
+            };
+          }>
+        ) => bus.putEvents($input.data.event),
+        () => {
+          return "success";
+        }
+      )
+  ).toThrow("object literal");
```
Comment (sam-goodwin) [inline test/api.test.ts:271]: Is this the whole message it throws?

**Review by sam-goodwin — APPROVED** [review]

File: src/event-bridge/event-bus.ts (hunk 1 above)
Comment (thantos) [reply src/event-bridge/event-bus.ts:386]: image

Looks like that is our miss

`export type PropertyName = Identifier | StringLiteral | NumericLiteral | ComputedPropertyName | PrivateIdentifier`

**Comment by thantos** [comment]
Rebased onto #347 to reflect the change before it gets pushed.

**Review by sam-goodwin — COMMENTED** [review]
Looks like there are changes brought across from my ast change? Merge problems?

**Pull request merged**